debugger/loadprinter.ml

```diff
@@ -117,8 +117,14 @@ let eval_value_path env path =
           let print_function ppf remote_val =
             print_with_fallback ppf (Obj.obj v ppf) remote_val in
           Printval.install_printer path ty_arg print_function
-      | Topprinters.Generic _ ->
-          raise (Error (`Wrong_type lid))
+      | Topprinters.Generic { ty_path ; arity } ->
+          let rec build v = function
+            | 0 -> Genprintval.Zero (fun ppf remote_val ->
+                print_with_fallback ppf (Obj.obj v ppf) remote_val)
+            | n ->
+                Genprintval.Succ
+                  (fun fn -> build ((Obj.obj v : _ -> Obj.t) fn) (n - 1)) in
```

Comment:
I have the impression that it could make sense to have the call to `Debugcom.Remote_value.repr` here, since it is the remote value protocol that decided to fetch the whole value on the debugger side.